### PR TITLE
swallow 400 as well for member check

### PR DESCRIPTION
**Pull request by honnix**
# Hey, I just made a Pull Request!

## Description

As commented in the code, we swallow 400 as well and move on to the next group.

## Motivation and Context

To not to fail too early.

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [x] All tests pass
- [x] Code coverage check passes
- [x] Error handling is tested
- [x] Errors are handled at the appropriate layer
- [x] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [x] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [x] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

**Comment by honnix** [comment]
Codecov is still complaining. Will check later.

File: styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java [hunk 1]
```diff
@@ -299,9 +299,13 @@ private boolean isMemberOfGroup(String principalEmail, String group) {
         return TRUE.equals(isMember);
       } catch (ExecutionException e) {
         final Throwable cause = e.getCause();
-        if (cause instanceof GoogleJsonResponseException
-            && ((GoogleJsonResponseException) cause).getStatusCode() == 404) {
-          return false;
+        if (cause instanceof GoogleJsonResponseException) {
+          final int statusCode = ((GoogleJsonResponseException) cause).getStatusCode();
+          // hasMember API returns 404 if the group does not exist, while returning 400 if the principal
+          // email does not exist or does not have the same domain as the group if it is enforced
```
Comment (danielnorberg) [inline styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java]: Feels unpleasant to silently swallow 400 bad request responses, as they could also be caused by legitimately broken requests or other problems. Let's at least log them.

And I'm curious about how we hit this case. How do user emails that do not exist make it through authentication? And do we have groups in our iam policies that do not belong to our domain?

Comment (honnix) [reply styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java]: The case we hit was for a legal service account, we check it against a group (with domain restrictions) having the role, and when that fails and we are not moving on to the next group.

Comment (honnix) [reply styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java]: It is being logged.

**Comment by Zatine** [comment]
👍 looks good to me

**Pull request merged**